iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveHvdcLine.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.modification.topology;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.modification.AbstractNetworkModification;
+import com.powsybl.iidm.network.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.powsybl.iidm.modification.topology.ModificationReports.*;
+
+/**
+ * @author Anis Touri <anis-1.touri@rte-france.com>
+ */
+public class RemoveHvdcLine extends AbstractNetworkModification {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RemoveHvdcLine.class);
+
+    private final String hvdcLineId;
+    private final List<String> shuntCompensatorsIds;
+
+    public RemoveHvdcLine(String hvdcLineId, List<String> shuntCompensatorsIds) {
+        this.hvdcLineId = Objects.requireNonNull(hvdcLineId);
+        this.shuntCompensatorsIds = Objects.requireNonNull(shuntCompensatorsIds);
+    }
+
+    @Override
+    public void apply(Network network, boolean throwException, ComputationManager computationManager, Reporter reporter) {
+        HvdcLine hvdcLine = network.getHvdcLine(hvdcLineId);
+        if (hvdcLine != null) {
+            HvdcConverterStation<?> hvdcConverterStation1 = hvdcLine.getConverterStation1();
+            HvdcConverterStation<?> hvdcConverterStation2 = hvdcLine.getConverterStation2();
+            hvdcLine.remove();
+            removedHvdcLineReport(reporter, hvdcLineId);
+            // Remove the Shunt compensators that represent the filters of the LCC
+            removeShuntCompensators(network, hvdcConverterStation1, hvdcConverterStation2, reporter);
+            removeConverterStations(hvdcConverterStation1, hvdcConverterStation2, reporter);
+        } else {
+            LOGGER.error("HVDC Line {} not found", hvdcLineId);
+            notFoundHvdcLineReport(reporter, hvdcLineId);
+            if (throwException) {
+                throw new PowsyblException("HVDC Line not found: " + hvdcLineId);
+            }
+        }
+    }
+
+    private void removeShuntCompensators(Network network,
+                                         HvdcConverterStation<?> hvdcConverterStation1,
+                                         HvdcConverterStation<?> hvdcConverterStation2,
+                                         Reporter reporter) {
+        if (!shuntCompensatorsIds.isEmpty()
+                && (isLccConverterStation(hvdcConverterStation1)
+                || isLccConverterStation(hvdcConverterStation2))) {
+
+            // Get Lcc voltage levels
```

Comment:
You can get 2 voltage levels with this :
VoltageLevel vl1 = hvdcLine.getConverterStation1().getTerminal().getVoltageLevel();
VoltageLevel vl2= hvdcLine.getConverterStation2().getTerminal().getVoltageLevel();